### PR TITLE
Add TiDB support

### PR DESCRIPTION
**Pull request by rleungx**
Add support for `CREATE DATASOURCE engine='tidb'`, including the example in docs.

### CODE REVIEW

File: mindsdb/integrations/handlers/tidb_handler/__about__.py [hunk 1]
```diff
@@ -0,0 +1,9 @@
+__title__ = 'MindsDB TiDB handler'
+__package_name__ = 'mindsdb_tidb_handler'
+__version__ = '0.0.1'
+__description__ = "MindsDB handler for TiDB"
+__author__ = 'MindsDB Inc'
```
Comment (ZoranPandovski) [inline mindsdb/integrations/handlers/tidb_handler/__about__.py:5]: Please add yourself as an author :)

**Review by ZoranPandovski — APPROVED** [review]
All looks good, please just add yourself as an author

**Comment by ZoranPandovski** [comment]
recheckcla

**Comment by mindsdbadmin** [comment]
**CLA Assistant Lite bot:** <br/>Thank you for your submission, we really appreciate it. Like many open-source projects, we ask that you sign our [Contributor License Agreement](https://github.com/mindsdb/mindsdb_native/blob/stable/assets/contributions-agreement/individual-contributor.md) before we can accept your contribution. You can sign the CLA by just posting a Pull Request Comment same as the below format.<br/>
   - - -
   ***I have read the CLA Document and I hereby sign the CLA***
   - - -
   <sub>You can retrigger this bot by commenting **recheck** in this Pull Request</sub>

**Comment by rleungx** [comment]
I have read the CLA Document and I hereby sign the CLA

**Review by ZoranPandovski — APPROVED** [review]

**Pull request merged**